### PR TITLE
Fix/6400 reject manipulated qr codes

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Add verification to Reject manipulated QR codes

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6400

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift [hunk 1]
```diff
@@ -83,6 +91,17 @@ class QRCodeVerificationHelper {
 			onError(CheckinQRScannerError.invalidCryptoSeed)
 			return
 		}
+		
+		let startTimeStamp = traceLocation.startDate?.timeIntervalSince1970 ?? 0
+		let endTimeStamp = traceLocation.endDate?.timeIntervalSince1970 ?? 0
+		let bothTimeStampsAreZero = (startTimeStamp == 0 && endTimeStamp == 0)
+		let startTimeIsBeforeEndTime = startTimeStamp < endTimeStamp
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift:98]: In our current implementation it is possible to set startDate and endDate to be equal. 
```suggestion
		let startTimeIsBeforeEndTime = startTimeStamp <= endTimeStamp
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift:98]: it says here it must be less, should i ignore it ?
Screenshot 2021-04-13 at 15 23 09
https://github.com/corona-warn-app/cwa-app-tech-spec/blob/f5ba0fdcaffc05e44df80067b8ec21d7ba721503/docs/spec/event-registration-client.md#attendee-check-in

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeVerificationHelper.swift:98]: I think it's best to ask Max about it, I am not aware of a restriction like that for the trace location generation. We would need to change the implementation there as well if this was the new requirement.

**Review by BLUM78691 — APPROVED** [review]
UA reviewed

**Review by nickguendling — APPROVED** [review]

**Pull request merged**